### ISSUE TITLE
Analyzer proposal: Add style check for public interface members

### ISSUE THREAD

**Issue by PaulBol** [DotNetAnalyzers/StyleCopAnalyzers#3586]
C# 8 adds access modifiers to interfaces. `public` [is the default](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods#detailed-design).

That means the following definitions are equivalent and may be used inconsistently.

```cs
public interface IStartable
{
    void Start();
}
```

```cs
public interface IStartable
{
    public void Start();
}
```

It would be nice if one variant could be enforced by StyleCop.

**Comment by PaulBol** [comment]
@sharwell - Thanks a lot for your input.

You are right. This should be doing what I had in mind:

```
dotnet_style_require_accessibility_modifiers = for_non_interface_members
dotnet_diagnostic.IDE0040.severity = error
```

So far I couldn't get it to work but I'll keep trying.

**Comment by PaulBol** [comment]
After some more tests with a minimal project I believe that IDE0040 does not do the job. Not all settings behave like I would expect.

Consider this class

```cs
    public class AcessModifierVariants
    {
        public int WithPublic { get; set; }
        internal int WithInternal { get; set; }
        private int WithPrivate { get; set; }
        int NoAccessModifier { get; set; }
    }
```

- With `dotnet_style_require_accessibility_modifiers = always` the property `NoAccessModifier` is detected
- With `dotnet_style_require_accessibility_modifiers = for_non_interface_members` the property `NoAccessModifier` is detected
- With `dotnet_style_require_accessibility_modifiers = omit_if_default` the property `WithPrivate` is detected
- **With `dotnet_style_require_accessibility_modifiers = never` no violations are detected**

Similarly for

```cs
public interface IStartable
{
    public void Start();
}
```

neither `dotnet_style_require_accessibility_modifiers = never` nor `dotnet_style_require_accessibility_modifiers = for_non_interface_members` create a violation.

I know it's sort of off-topic for StyleCop but can you suggest if I'm doing something wrong or observing a bug?

**Comment by bjornhellander** [comment]
I see some unexpected behavior as well. If this is still relevant, I would suggest you ask over at https://github.com/dotnet/roslyn/issues instead.
